integration-tests/bats/import-mysqldump.bats

```diff
@@ -0,0 +1,241 @@
+/*!40000 ALTER TABLE mytable ENABLE KEYS */;
+UNLOCK TABLES;
+
+/*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
+/*!50003 SET sql_mode              = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION' */ ;
+DELIMITER ;;
+/*!50003 CREATE*/ /*!50017 DEFINER=root@localhost*/ /*!50003 TRIGGER tt BEFORE INSERT ON mytable FOR EACH ROW SET NEW.v1 = NEW.v1 * 11 */;;
+DELIMITER ;
+/*!50003 SET sql_mode              = @saved_sql_mode */ ;
+SQL
+    [ "$status" -eq 0 ]
+
+    dolt sql -q "INSERT INTO mytable VALUES (6,8);"
+    run dolt sql -q "SELECT * FROM mytable;"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "| 6  | 88 |" ]] || false
+}
+
+@test "import-mysqldump: import a database with procedure dumped with --routines flag" {
+    run dolt sql <<SQL
+/*!50003 DROP PROCEDURE IF EXISTS new_proc */;
+/*!50003 SET @saved_sql_mode       = @@sql_mode */ ;
+/*!50003 SET sql_mode              = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION' */ ;
+DELIMITER ;;
+CREATE DEFINER=root@localhost PROCEDURE new_proc(x DOUBLE, y DOUBLE)
+SELECT x*y ;;
+DELIMITER ;
+/*!50003 SET sql_mode              = @saved_sql_mode */ ;
+SQL
+    [ "$status" -eq 0 ]
+
+    run dolt sql -q "CALL new_proc(2, 3);"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "| 6       |" ]] || false
+}
+
+@test "import-mysqldump: a table with string literal representation in column definition" {
+    skip "charset introducer needs to be supported in LIKE filter"
+    run dolt sql <<SQL
+CREATE TABLE mytable (
+  pk int NOT NULL,
+  col2 int DEFAULT (date_format(now(),_utf8mb4'%Y')),
+  col3 varchar(20) NOT NULL DEFAULT 'sometext',
+  PRIMARY KEY (pk),
+  CONSTRAINT status CHECK ((col3 like _utf8mb4'%sometext%'))
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+SQL
+    [ "$status" -eq 0 ]
+
+    dolt sql -q "INSERT INTO mytable VALUES (1, 2003, 'first_sometext');"
+    run dolt sql -q "SELECT * FROM mytable;"
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "| 1  | 2003 | first_sometext |" ]] || false
+}
+
+@test "import-mysqldump: some of tables from mysql db, dump with --all-databases flag" {
+    skip "utf8mb3 charset introducer needs to be supported for mysql db tables"
+    run dolt sql <<SQL
+CREATE TABLE engine_cost (
+  engine_name varchar(64) NOT NULL,
+  device_type int NOT NULL,
+  cost_name varchar(64) NOT NULL,
+  cost_value float DEFAULT NULL,
+  last_update timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+  comment varchar(1024) DEFAULT NULL,
+  default_value float GENERATED ALWAYS AS ((case cost_name when _utf8mb3'io_block_read_cost' then 1.0 when _utf8mb3'memory_block_read_cost' then 0.25 else NULL end)) VIRTUAL,
+  PRIMARY KEY (cost_name,engine_name,device_type)
+) /*!50100 TABLESPACE mysql */ ENGINE=InnoDB DEFAULT CHARSET=utf8mb3 STATS_PERSISTENT=0 ROW_FORMAT=DYNAMIC;
+SQL
```

Comment:
Would love a show create table here in the future. I know this doesn't work right now.